build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -94,38 +100,79 @@ class AssetGraph {
   /// set of assets that need to be deleted.
   Iterable<AssetId> updateAndInvalidate(Map<AssetId, ChangeType> updates) {
     var deletes = new Set<AssetId>();
-    var seen = new Set<AssetId>();
     void clearNodeAndDeps(AssetId id, ChangeType rootChangeType,
-        {AssetId parent}) {
-      if (seen.contains(id)) return;
-      seen.add(id);
+        {AssetId parent, bool rootIsSource}) {
```

Comment:
We report deletes, not edits

https://github.com/dart-lang/build/blob/7820ae1b143068a4d75bf35029cbd90a429f690c/build_runner/lib/src/generate/watch_impl.dart#L177

Maybe we should not report deletes either?